aeron-samples/src/main/java/uk/co/real_logic/aeron/samples/AeronStat.java

```diff
@@ -29,31 +29,44 @@
 import java.util.concurrent.atomic.AtomicBoolean;
 
 /**
- * App to print out status counters and labels
+ * Application to print out status counters and labels
+ * A command and control memory file is maintained by media driver in shared memory.
+ * This application reads the the cnc file and prints the statistics counters.
+ * Layout of the cnc file is described in {@link CncFileDescriptor}.
  */
 public class AeronStat
 {
     public static void main(final String[] args) throws Exception
     {
+        // Get command and control file path
         final File cncFile = CommonContext.newDefaultCncFile();
 
+        // Print command and control file path
         System.out.println("Command `n Control file " + cncFile);
 
+        // Check that command and control file exists, open file, and return MappedByteBuffer for entire file
         final MappedByteBuffer cncByteBuffer = IoUtil.mapExistingFile(cncFile, "cnc");
 
+        // Create a DirectBuffer for cnc's metadata
         final DirectBuffer metaDataBuffer = CncFileDescriptor.createMetaDataBuffer(cncByteBuffer);
 
+        // Extract version number of cnc from meta data
         final int cncVersion = metaDataBuffer.getInt(CncFileDescriptor.cncVersionOffset(0));
+
+        // Throw an exception if CNC version number mismatch is detected between this app and memory mapped cnc file
```

Comment:
Really? Is a comment needed for things like this? I'd prefer no comments in a code base unless it is genuinely tricky to describe in code or referencing an external algorithm or paper.